### PR TITLE
docs: update PR template

### LINKED ISSUE #3839: [CLI Quality] Modify docs reflecting updates about PR Review
The team working on the CLI has been making agreements and creating expectations for each other when we contribute and review changes of this repo. As a follow-up action, we agree to update the contributing guidelines of the repo.

Additionally, we should consider updating the PR Template to remind us about such agreements and help other discover them.

As inspiration, we can look at the following:
- https://github.com/embeddedartistry/templates/blob/master/oss_docs/PULL_REQUEST_TEMPLATE.md
- https://gist.github.com/paseaf/b7baed4f162e5aba5827c793c4355bed

### PR DESCRIPTION
**Pull request by andreafalzetti**
Fixes #3839

This PR updates the PR template reflecting our internal agreements and expectations.

## How to validate

N/A

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+
+## CLI Quality Reminders 🔧
+
+For both authors and reviewers:
+
+- Scrutinize for potential regressions
+- Ensure key automated tests are in place
+- Build the CLI and test using the validation steps
+- Assess Developer Experience impact (log messages, performances, etc)
+- If too broad, consider breaking into smaller PRs
```
Comment (AdrianPedriza) [inline .github/pull_request_template.md:23]: link here https://github.com/okteto/okteto/blob/master/docs/code-review.md#pr-scope I think could be a good idea

Comment (andreafalzetti) [reply .github/pull_request_template.md:23]: good shout, I will move the PR to draft and review that one too 👍

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,5 +1,35 @@
-# Proposed changes
+# Description
```
Comment (teresaromero) [inline .github/pull_request_template.md:1]: Maybe we can remove this title and go strainght with the Fixes header? 

for title i would rather use something more descriptive

Comment (andreafalzetti) [reply .github/pull_request_template.md]: I've reverted to `Proposed changes` as I think it's more inclusive of "fixes" + "desc" - I wanted to highlight the "description" part but we always add it, to it's not important to highlight it

**Review by teresaromero — APPROVED** [review]

**Pull request merged**